test-data/unit/check-based-intersection.test

```diff
@@ -0,0 +1,77 @@
+[case testIntersection]
+from basedtyping import Intersection
+i: Intersection[int, str]
+reveal_type(i)  # N: Revealed type is "int & str"
+[builtins fixtures/tuple.pyi]
+
+
+[case testIntersectionSyntax]
+from __future__ import annotations
+i: int & str
+reveal_type(i)  # N: Revealed type is "int & str"
+[builtins fixtures/tuple.pyi]
+
+
+[case testNoIntersectionSyntax]
+i: int & str  # E: X & Y syntax for intersections requires quoted types or __future__.annotations  [valid-type]
+
+
+[case testIntersectionAttribute]
+from __future__ import annotations
+class A:
+    a: int
+
+class B:
+    b: str
+
+i: A & B
+reveal_type(i.a)  # N: Revealed type is "int"
+reveal_type(i.b)  # N: Revealed type is "str"
+reveal_type(i.c)  # E: "A & B" has no attribute "c"  [attr-defined] \
+                  # N: Revealed type is "Any (from error)"
+[builtins fixtures/tuple.pyi]
+
+
+[case testIntersectionAssignability]
+# flags: --disallow-redefinition
```

Comment:
why enable `--disallow-redefinition`? this test doesnt seem to need it, or is this part of the test?

Reply:
I don't want the top level assignments in this test to change the types of the variables.

Reply:
wait top level assignments count as redefinitions even when you dont explicitly give it a type annotation? that sounds really sus

Reply:
there's an issue for it